src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -3,957 +3,1496 @@
-			shaders.fragment = createShader(spirv.fragment, IShader::E_SHADER_STAGE::ESS_FRAGMENT);
-		}
-	
-		SVertexInputParams vertexInputParams{};
-		{
-			vertexInputParams.enabledBindingFlags = 0b1u;
-			vertexInputParams.enabledAttribFlags = 0b111u;
-
-			vertexInputParams.bindings[0].inputRate = asset::SVertexInputBindingParams::EVIR_PER_VERTEX;
-			vertexInputParams.bindings[0].stride = sizeof(ImDrawVert);
-
-			auto& position = vertexInputParams.attributes[0];
-			position.format = EF_R32G32_SFLOAT;
-			position.relativeOffset = offsetof(ImDrawVert, pos);
-			position.binding = 0u;
-
-			auto& uv = vertexInputParams.attributes[1];
-			uv.format = EF_R32G32_SFLOAT;
-			uv.relativeOffset = offsetof(ImDrawVert, uv);
-			uv.binding = 0u;
-
-			auto& color = vertexInputParams.attributes[2];
-			color.format = EF_R8G8B8A8_UNORM;
-			color.relativeOffset = offsetof(ImDrawVert, col);
-			color.binding = 0u;
-		}
+	if (!layouts[texturesInfo.setIx])
+		return nullptr;
 
-		SBlendParams blendParams{};
-		{
-			blendParams.logicOp = ELO_NO_OP;
+	if (!layouts[samplersInfo.setIx])
+		return nullptr;
 
-			auto& param = blendParams.blendParams[0];
-			param.srcColorFactor = EBF_SRC_ALPHA;//VK_BLEND_FACTOR_SRC_ALPHA;
-			param.dstColorFactor = EBF_ONE_MINUS_SRC_ALPHA;//VK_BLEND_FACTOR_ONE_MINUS_SRC_ALPHA;
-			param.colorBlendOp = EBO_ADD;//VK_BLEND_OP_ADD;
-			param.srcAlphaFactor = EBF_ONE_MINUS_SRC_ALPHA;//VK_BLEND_FACTOR_ONE_MINUS_SRC_ALPHA;
-			param.dstAlphaFactor = EBF_ZERO;//VK_BLEND_FACTOR_ZERO;
-			param.alphaBlendOp = EBO_ADD;//VK_BLEND_OP_ADD;
-			param.colorWriteMask = (1u << 0u) | (1u << 1u) | (1u << 2u) | (1u << 3u);//VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT;
+	return device->createPipelineLayout(PushConstantRanges, std::move(layouts[0u]), std::move(layouts[1u]), std::move(layouts[2u]), std::move(layouts[3u]));
+}
 
-		}
-	
-		SRasterizationParams rasterizationParams{};
-		{
-			rasterizationParams.faceCullingMode = EFCM_NONE;
-			rasterizationParams.depthWriteEnable = false; // TODO: check if it disabled depth test?
-			rasterizationParams.depthBoundsTestEnable = false;
-			// rasterizationParams.stencilTestEnable = false; // TODO: check if stencil test is disabled
-		}
+const smart_refctd_ptr<IFileArchive> UI::mount(smart_refctd_ptr<ILogger> logger, ISystem* system, const std::string_view archiveAlias)
+{
+	assert(system);
+		
+	if(!system)
+		return nullptr;
 
-		SPrimitiveAssemblyParams primitiveAssemblyParams{};
-		{
-			primitiveAssemblyParams.primitiveType = EPT_TRIANGLE_LIST;
-		}
+	auto archive = make_smart_refctd_ptr<builtin::CArchive>(smart_refctd_ptr(logger));
+	system->mount(smart_refctd_ptr(archive), archiveAlias.data());
 
-		{
-			const IGPUShader::SSpecInfo specs[] =
-			{
-				{ .entryPoint = "VSMain", .shader = shaders.vertex.get() },
-				{ .entryPoint = "PSMain", .shader = shaders.fragment.get() }
-			};
+	return smart_refctd_ptr(archive);
+}
```

Comment:
Needs to be a counterpart to https://github.com/Devsh-Graphics-Programming/Nabla/blob/209d6006ec59503b733a9db48e1eff8aadc3135a/include/nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp#L36

Reply:
> if there's no archive to be mounted

actually the way it works now makes this scenario impossible, there is no dependency between `NBL_EMBED_BUILTIN_RESOURCES` and the archive is always shipped within the extension - I find it a case since its "extension", all utilities should be provided within, including the archive imho

Reply:
> so the folder gets mounted if there's no archive to be mounted

resolved, but see [this](https://github.com/Devsh-Graphics-Programming/Nabla/pull/707/commits/c2113466ea0e4345bf4e5d079ccd6f0c7a5d20da#diff-7a62dc4d96706dca948adc887e6c8b5b6c2dab1adc35738156389f013c7d78c8R125) one please